xen/arch/arm/tee/ffa_partinfo.c

```diff
@@ -176,9 +193,75 @@ int32_t ffa_handle_partition_info_get(uint32_t w1, uint32_t w2, uint32_t w3,
         }
     }
 
-    *count = num_ffa_sp;
+    if ( ffa_fw_support_fid(FFA_PARTITION_INFO_GET) )
+    {
+        ffa_hyp_rx_release();
+        spin_unlock(&ffa_rx_buffer_lock);
+    }
+
+    *count = num_ffa_sp + num_ffa_vm;
     *size = dst_size;
 
+    /* add the VM informations if any */
+    if ( !num_ffa_vm )
+        return FFA_RET_OK;
+
+    rcu_read_lock(&domlist_read_lock);
+    for_each_domain( dom )
+    {
+        struct ffa_ctx *vm = dom->arch.tee;
+
+        /*
+         * we do not add the VM calling to the list and only VMs with
+         * FF-A support
+         */
+        if (dom != d && vm != NULL && vm->guest_vers != 0)
+        {
+            /*
+             * We do not have UUID info for VMs so use
+             * the 1.0 structure so that we set UUIDs to
+             * zero using memset
+             */
+            struct ffa_partition_info_1_0 srcvm;
+
+            if ( !num_ffa_vm )
+            {
+                /*
+                 * The number of domains changed since we counted them, we
```

Comment:
If a domain is removed instead, we'll report a too large value in *count.
How about counting the domains in this loop instead when FFA_PARTITION_INFO_GET_COUNT_FLAG isn't supplied?

Reply:
I am using this upper to check the amount of memory available but definitely i need to fix the case where a partition vanished and fix count.